src/cors.ts

```diff
@@ -9,15 +9,29 @@ export function setupCors(
 ) {
   const corsDebug = createDebug('signalk-server:cors')
 
+  const corsOptions: CorsOptions = {
+    credentials: true
+  }
+
   const corsOrigins = allowedCorsOrigins
     ? allowedCorsOrigins
         .split(',')
         .map((s: string) => s.trim().replace(/\/*$/, ''))
     : []
-  corsDebug(`corsOrigins:${corsOrigins.toString()}`)
-  const corsOptions: CorsOptions = {
-    credentials: true,
-    origin: corsOrigins
+
+  // default wildcard cors configuration does not work
+  // with credentials:include client requests, so add
+  // our own wildcard rule that will match all origins
+  // but respond with that origin, not the default *
+  if (allowedCorsOrigins?.startsWith('*')) {
+    corsOptions.origin = (origin: string | undefined, cb) => cb(null, origin)
```

Comment:
Wouldn't it be better instead of doing this to set corsOptions.credentials to false instead?

Reply:
The use case is cors requests with credentials sent by the client - supporting authenticated cors requests. Without credentials true that won’t work.

Reply:
Hmm, ok. We are basically solving the client-issue by allowing anything that comes in. Feels odd, but understand the solution.

Reply:
Only if you enter *. You can still limit by origin or disallow credentials by default config with empty allowedOrigins.